pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -1020,9 +1021,9 @@ func (cb *ClusterBuilder) buildUpstreamClusterTLSContext(opts *buildClusterOpts,
 				ValidationContextSdsSecretConfig: authn_model.ConstructSdsSecretConfig(authn_model.SDSRootResourceName),
 			},
 		}
-		// Set default SNI of cluster name for istio_mutual if sni is not set.
+		// Set default SNI for istio_mutual if sni is not set.
 		if len(tlsContext.Sni) == 0 {
-			tlsContext.Sni = c.cluster.Name
+			tlsContext.Sni = opts.istioMtlsSni
```

Comment:
istioMtlsSni can be empty and what's the behavior of envoy. Does it send no sni?

Reply:
yes. It does. But we should not use cluster name here though - it can be different from istioMTLSSni